pkg/local_instance.go

```diff
@@ -827,12 +832,31 @@ func (li LocalInstance) RunModesString() string {
 	return strings.Join(lo.Uniq[string](result), ",")
 }
 
+func (li LocalInstance) passwordFile() string {
+	return fmt.Sprintf("%s/crx-quickstart/conf/admin-password.properties", li.Dir())
+}
+
+func (li LocalInstance) needsPasswordFile() bool {
+	return !li.IsInitialized() && li.instance.password != instance.PasswordDefault
+}
+
+func (li LocalInstance) savePasswordFile() error {
+	if li.needsPasswordFile() {
+		return filex.WriteString(li.passwordFile(), fmt.Sprintf("admin.password=%s", li.instance.password))
+	}
+	return nil
+}
+
 func (li LocalInstance) JVMOptsString() string {
 	result := append([]string{}, li.JvmOpts...)
 
 	// at the first boot admin password could be customized via property, at the next boot only via Oak Run
-	if !li.IsInitialized() && li.instance.password != instance.PasswordDefault {
-		result = append(result, fmt.Sprintf("-Dadmin.password=%s", li.instance.password))
+	if li.needsPasswordFile() {
+		if pathx.Exists(li.passwordFile()) {
+			result = append(result, fmt.Sprintf("-Dadmin.password.file=%s", li.passwordFile()))
+		} else {
+			log.Fatalf("%s > cannot find password file '%s'", li.instance.ID(), li.passwordFile())
```

Comment:
find indicates searching, but you know where file is located;
it's rather `required password file does not exist at path 'xx'`